README.md
WYSIWYG-CKeditor
===
This is modified extension of MediaWiki: WYSIWYG -editor. This bundle includes WYSIWYG extension and other modifed components related to it.
Files of this bundle were originally taken from diffrent source bundles of extension WYSIWYG from MediaWiki site www.mediawiki.org.
You can find results of this work here (http://github.com/Mediawiki-wysiwyg).

More information about MediaWiki extension WYSIWYG can be found here:

  http://www.mediawiki.org/wiki/Extension:WYSIWYG

  http://www.mediawiki.org/wiki/Extension_talk:WYSIWYG

-----------
**NOTE!**

> You will use this bundle in your environment at your own risk.  The authors of this package can not be held responsible for any issues this bundle may cause in your system.

-----------

> It is always nice to hear comments about use of this extension. You can leave your comments here:

>http://www.mediawiki.org/wiki/Extension_talk:WYSIWYG

----------
**You can find information about following issues below:**
- history of modifications (in reverse order)
- supported languages
- version info of compatible components required with this extension and Mediawiki
- installation instructions (MediaWiki, WYSIWYG, WikiEditor, SemanticForms)
- compatible browsers and browser settings

================

------------
History of modifications:
===

- 30.10.15  Modifications with table captions: added attribute support, fixed error which cased WYSIWYG to crash with table caption. Version 1.5.6_0 [B551+30.10.2015].

- 25.10.15  Updated version of CKeditor source files to version 4.5.4 [revision d4677a3]. Moved source- files of CKeditor-WYSIWYG into ckeditor_source- directory, removed old plugins and other files, updated compile-minify instructions. Fixed error of previous WYSIWYG versions from [B551+31.01.2015] to [B551+19.10.2015] which prevented WYSIWYG to start in source mode. Version 1.5.6_0 [B551+25.10.2015].

- 19.10.15  Update of CKEditor from version 4.5.1 [revision a513a92] to CKeditor 4.5.4 [revision d4677a3] (_source directory was not yet updated). In About- dialog of CKeditor, show also info about version of WYSIWYG. Version 1.5.6_0 [B551+19.10.2015].

- 01.09.15  Show names of categories in wysiwyg mode (by Varlin 28.07.15). Fixed double click of category element and tooltip of category button in menu. Workaround to avoid manual modifications because of "protected property OutputPage". Version 1.5.6_0 [B551+01.09.2015].

- 07.07.15  Fixed in link dialog: when editing existing external link without separate display text (just save, no modifications), display text was changed to something like this "[1]". Version 1.5.6_0 [B551+07.07.2015].

- 06.07.15  Update of CKEditor from version 4.4.5 [revision 25cdcad] to CKeditor 4.5.1 [revision a513a92]. Version 1.5.6_0 [B551+13.03.2015].

- 06.07.15  Created branch CKeditor_v45 for update of CKeditor v4.5.1 [revision a513a92]. Version 1.5.6_0 [B551+13.03.2015].

- 13.03.15  Merge pull request #64 from Varlin/patch-27: French translation. Version 1.5.6_0 [B551+13.03.2015].

- 11.03.15  Created branch CKeditor_V4_Test for easier management of merged contributions. Fixed display of right version number. Version 1.5.6_0 [B551+07.03.2015].

- 10.03.15  Merged modifications of branch Syntaxhighlight-Nowiki-Pre into branch CKeditor_v4. Version 1.5.6_0 [B551+07.03.2015].

- 07.03.15  Modified place of class attribute "fck_mv_" with img-elements to be last one (tag:06.03.15). Version 1.5.6_0 [B551+07.03.2015].

- 06.03.15  Allow every kind of templates (batch-18 #51). Fix <caption> support ("titles" of tables, batch-20 #55). Set "$wgDefaultUserOptions['riched_use_popup'] = 0;" by default (batch-23 #58). Avoid getting an upper case to selflink, do not apply to category/property (batch-21 #56). Modified enabling/disabling link buttons (batch-22 #57).  All prev. modications by Varlin (tag:06.03.15). Version 1.5.6_0 [B551+06.03.2015].

- 05.03.15  Wysiwyg does not support IE versions below or eq. to IE8. Fixed double click error of pre- element when "$wgFCKEditorSpecialElementWithTextTags = 1". Fixed test when wikitext to html conversion can be started "editorSrcToWswTrigger = true". Removed one br- tag which caused extra line feeds when toggling source-wysiwyg mode. Version: 1.5.6_0 [B551+05.03.2015], branch CKeditor_v4_Nowiki-Synt-Pre (tags: Syntaxhighlight-Nowiki-Pre, "03.03.15").

- 26.02.15  Edit texts inside nowiki and syntaxhighlight tags directly on page: removed unnecessary html decode/encode calls. Version: 1.5.6_0  [B551++26.02.2015]. Branch CKeditor_v4_Nowiki-Synt-Pre (Syntaxhighlight-Nowiki-Pre).

- 26.02.15  Edit texts inside nowiki and syntaxhighlight tags directly on page: toolbar button to create/remove nowiki/syntaxhighlight format; variable "$wgFCKEditorSpecialElementWithTextTags = 1" in LocalSettings.php to get img- element with nowiki/syntaxhighlight tags. Icon of toolbar button for template- dialog changed. Version: 1.5.6_0  [B551+26.02.2015]. Branch CKeditor_v4_Nowiki-Synt-Pre (Syntaxhighlight-Nowiki-Pre).

- 23.02.15  Added language variables for link- and template- dialogs (by Varlin). Modified translations: fi, de. Version: 1.5.6_0  [B551+23.02.2015].

- 10.02.15  Fixed activation of undo/redo buttons when switching from source to wysiwyg view.  Version: 1.5.6_0 [B551+10.02.2015].

- 08.02.15  Show paragraph format "Formatted" with light grey background. Version: 1.5.6_0 [B551+08.02.2015].

- 05.02.14  Created branch "CKeditor_v4_preTag" for tests of pre -tag etc and merged modifications of pre- tag (by rchouine). Fixed text corruption with sequential pre- tags. Use special element with pre- tags in case $wgFCKEditorSpecialElementWithPreTag = 1 in LocalSettings.php. Fixed extra line break with magic words like __TOC__ etc. Version: 1.5.6_0 [B551+05.02.2015].

- 04.02.15  Merged branch "CKeditor_v4_HtmlComments" to "CKeditor_v4". Version: 1.5.6_0 [B551+02.02.2015].

- 02.02.15  Fix with new category dialog in case name of category contains double quotes. Show text "Double-click to edit the value" with editable elements in wysiwyg mode. Show link target as tooltip with external links. Show text of caption as tooltip with images. Version: 1.5.6_0 [B551+02.02.2015].

- 31.01.15  Removed modification "disabled unknown object message of CKeditor" which is not needed after new category dialog. Version: 1.5.6_0 [B551+31.01.2015].

- 24.01.15  Created new branch "CKeditor_v4_HtmlComments": fix for html comment -tags when commented block contains incomplete html code, catch exceptions in _getNodeFromHtml; fix for location of image file when defining original size of image; disabled unknown object message of CKeditor; fix for spaces in external link. Version: 1.5.6_0 [B551+24.01.2015].

- 24.01.15  Merged branch "CKeditor_v4_Category" to "CKeditor_v4" (=new dialog for category definitions). Version: 1.5.6_0 [B551+23.01.2015].

- 23.01.15  Fixed doubleclick problem with selection lists in new dialog of categories (by DavidBeaumier). Version: 1.5.6_0 [B551+23.01.2015].

- 22.01.15  Merged tooltip modifications of new category -dialog (by DavidBeaumier). Version: 1.5.6_0 [B551+22.01.2015].

- 20.01.15  Merged modifications of new category -dialog (by DavidBeaumier). Added support for selected user language. Version: 1.5.6_0 [B551+20.01.2015].

- 19.01.15  Created new branch "CKeditor_v4_Category" for modifications of new category dialog by DavidBeaumier.

- 12.01.15  Modified wikitext to html conversions so that toHtml function is not called unnecessarily two times (with page is opening, source button and toggle link). Keep editor in read-only mode and buttons of wysiwyg menu disabled during wysiwyg-wikitext conversion in both directions (this has visible effect with big wiki pages, when conversion takes long time). Added partial support of text "Page is loading, please wait...". Version: 1.5.6_0 [B551+12.01.2015].

- 09.01.15  Updated WYSIWYG\ckeditor\_source directory. Version: 1.5.6_0 [B551+09.01.2015].

- 04.01.15  Modification related to formats of link texts (old "B551+27.12.2014"). Version: 1.5.6_0 [B551+04.01.2015].

- 02.01.15  Cancelled modification related to formats of link texts (old "B551+27.12.2014"), because it caused problems with html related characters '&', '"', '\'', '<', '>' which were used in other texts than links. Version: 1.5.6_0 [B551+02.01.2015].

- 31.12.14  Modifications with image dialog: canceled use of default size with images; width and height can both be specified, only one of them or none of them; size definition "upright"; horizontal alignment "none"; vertical alignments "baseline, sub, super, top, text-top, middle, bottom, text-bottom"; link target (external or internal); disable link. Merged some french translations (by Varlin). Version: 1.5.6_0 [B551+31.12.2014].

- 30.12.14  Fixed resizing of image which did not always work with FF (related to merged modification of batch-9 "Fix resizing problem in Firefox #17" at 27.08.14). If size has not been given in dialog, use (and display) default size of the image. Version: 1.5.6_0 [B551+30.12.2014].

- 27.12.14  Fixed use of formats italic, bold and undeline inside display text of internal/external link. Version: 1.5.6_0 [B551+27.12.2014].

- 23.12.14  Fixed handling of image link in case image file is not yet available in MW and link is edited in wysiwyg (=noimage.png is always considered internal link). Version: 1.5.6_0 [B551+++23.12.2014].

- 23.12.14  Fixed clearing preview of image (uploading of new image failed). Version: 1.5.6_0 [B551++23.12.2014].

- 23.12.14  When existing image is double clicked, display preview of image in image dialog. Version: 1.5.6_0 [B551+23.12.2014].

- 22.12.14  Fixed issue with caption: in case existing image was double clicked, caption was lost (by Aarakast). Version: 1.5.6_0 [B551+22.12.2014].

- 12.12.14  Updated version info: 1.5.6_0 [B551+12.12.2014]

- 12.12.14  Fixed handling of sort key with categories: earlier spaces in category name caused sort key to be name of category, now sort key obeys name of page by default.

- 11.12.14  Fixed French translation (by Varlin, commit b5bc0784af2b4e481d6195439a733ed78bce8874 [b5bc078]). Fixed $wgGroupPermissions in sample LocalSettings.php below (by nemobis).

- 05.12.14  Fixed handling of some special characters (f.ex: '&' and '"') with category -dialog (character '_' will still be converted to space).

- 26.11.14  Added possibility to use mediawiki variables, f.ex ROOTPAGENAME, with internal links. Added new variable names and magic words of latest MW versions.

- 25.11.14  Fixed handling of empty lines inside numbered and bulleted lists.

- 23.11.14  Fixed tag- dialog (try to parse any simple tags of wikitext unknown to wysiwyg): pass name of attributes of wikitags to special.js using own special attribute (_fck_mw_tagattributes); removed extension specific predefined attribute list; do not use tag names of MW in tag -dialog (except converting source -tag to syntaxhighlight -tag).

- 21.11.14  It is possible to enter also template definition with tag -dialog (template dialog, template.js, is preferred method to work with template definitions).

- 21.11.14  Fixed data of gallery -tag to be edited with tag -dialog (tag is displayed as gallery -element in wysiwyg). Enabled use of attributes for hovergallery -tag (tag is displayed as special -element in wysiwyg).

- 20.11.14  Fixed tag- dialog: support comments -tag (about 99% compatible); try to parse any simple tags of wikitext unknown to wysiwyg.

- 20.11.14  Fixed tag- dialog: allow more advanced tag definitions in tag -dialog; reactivated text-html-text conversions using browser (related to fix dated 19.11.14 below); use tag -dialog to edit math -formulas. Changed handling of math -element from img- to span- tag making it identical with other similar elements of wysiwyg.

- 19.11.14  Fixed tag- dialog: support multiple nested tags of wikitext or html syntax by converting inserted text to html format (related to fix dated 30.10.14 below).

- 17.11.14  Modified wysiwyg to work with poll- extension (by default all unkown simple wikitext tags should now work as "special" -element).

- 07.11.14  Updated CKBuilder and run instructions of _source directory. Fixed CKBuilder warnings with smwtoolbar/plugin.js. Added command _source/dev/builder/build_leave-js-unminified.sh.

- 07.11.14  Fixed version info of CKeditor on page "Special:Version" (CKeditor 4.4.5 revision 25cdcad). Updated CKeditor source related files in ckeditor/_source from githhub repository of CKeditor. Added source code of texttransform plugin. Instructions and sample config.js to run source files of CKeditor with WYSIWYG (_source/_ReadMe-ckeditor_source.txt).

- 07.11.14  Updated version of CKeditor from 4.4.4 to 4.4.5 (earlier 23.08.14 "default table properties" modification with table.js was cancelled by this update, textselection plugin has not been updated). Replaced contents of ckeditor\_source directory with source files of CKEditor 4.4.5. After this copied following directories from directory ckeditor\plugins to directory ckeditor\_source\plugins: mediawiki, mwtemplate, smwqueryinterface, smwrichmedia, smwrule, smwtoolbar, smwwebservice, textselection.

- 04.11.14  Modified IE compatibility mode setting "IE=9" to be the first of all metas (like it should be, related to mod. made at 17.02.13).

- 02.11.14  With syntaxhighlight extension use tag syntaxhighlight by default, source tag is still supported (f.ex xml  tag source works now). Convert existing "old" source -tag on page or keyed in source -tag in dialog special.js to syntaxhighlight -tag.

- 30.10.14  With dialog special.js modified html-txt-html conversion, because it was too strict for the purpose of dialog.

- 30.10.14  Fixed behaviour of source -object with syntaxhighlight extension: convert existing "old" syntaxhighlight -tag on page (code by Wingsofcourage) or keyed in syntaxhighlight -tag in dialog special.js to source -tag; with code samples keep line indents created by spaces (fckSPACE); convert some of the html tags to text and vice versa when editing code samples (dialog special.js).

- 21.10.14  Modified test of compatible browser string for IE (code provided by interjinn).

- 20.10.14  Fixed handling of external link in case it has format [//www.xx.yy].

- 17.10.14  Image of math equations "button_math.png" may be in different locations, modified to test location of MW in skins/common/images and extensions/Math/images.

- 08.10.14  Cancelled modification dated 23.08.14 below "Merge pull request #16 from Varlin/patch-8, Fix unwanted caps" (commit: 83b81d5ca2a6493a72f41c709f1dbdfd4bec5bb7 [83b81d5]) because it corrupted definitions of sortkey with existing category defintions on page.

- 07.10.14  WYSIWYG extension is compatible with MW 1.22.0 - 1.22.12 and MW 1.23.0 - 1.23.5. Updated Preferences=>Editing section in this README.MD file.

- 29.09.14  Fixed missing German translations of image -dialog: imgTypeThumb, imgTypeFrame, imgTypeFrameless, imgTypeBorder.

- 12.09.14  Textselection -plugin works in direction wysiwyg->wikitext with FF, Chrome and IE in MW 1.22.x. In MW 1.23.x it works too, but IE does not scroll to selected location automatically, you have to do it manually. Removed previous IE fix below dated 11.09.14.

- 11.09.14  Upgraded WikiEditor v0.3.1 (of MW 1.22.0) to v0.4.0 (of MW 1.23.3).

- 11.09.14  In MW 1.22.10 + IE11 textselection plugin in wysiwyg->wikitext direction does not work. Because extension is enabled and wysiwyg->wikitext direction works with FF and Chrome, bookmarks are left in wikitext code with IE, so remove these: `<span data-cke-bookmark="1" style="display: none;" id="cke_bm_190c">&nbsp;</span>`. In case texselection plugin is fixed to work with MW and IE then this should be removed. MW1.23.3 + IE11 does not have this bookmark problem.

- 10.09.14  Problems in MW 1.23.3 with IE, FF and Chrome to pass selected text to functions SimpleLink, Category and Ref properly => use function getSelectedText() in order to avoid browser issues (works with IE8-IE11, FF, Chrome). Textselection plugin does not work with MW 1.23.3 and IE 11

- 09.09.14  Fixed version info of CKeditor for page "Special:Version" (CKeditor 4.4.4 revision 1ba5105).

- 08.09.14  Fixed dialog of categories: handling of spaces in category names.

- 08.09.14  Updated CKeditor from version 4.4.0 (revision 98daef5) to version 4.4.4 (revision 1ba5105). Added ckeditor plugins: textransform, tableresize, and textselection. NOTES ABOUT PLUGINS! 1. Textransform works ok; 2. Tableresize works too but inserts html code when adjusting width of column; 3. Textselection plugin works with FF and Chrome, it has also been hacked to work only in wysiwyg->wikitext direction because other direction left bookmarks on page; 4. Default properties of table -plugin have been modified to no border - one row - width = 100%. It is not very good idea to hack plugins which are delivered by CKeditor distribution, because hacking makes upgrade of CKeditor difficult!

- 06.09.14  Fixed enable/disable of "create link" -button. Added parameter in user editing preferences "Paste selected text as link text into link -dialog", modified link -dialog to allow pasting selected text as name of link (based on Varlin's "patch-10", modified further).

- 05.09.14  Added one button to convert plain text to simple link (based on Varlin's "patch-11", modifed to work with IE, new icon for button of simple link, fixed enable/disable of link buttons): config.js, plugin.js, tb_icon_simplelink.gif. Due to user error earlier, some language files of ckeditor have not been included, files were added now.

- 29.08.14  With search function used in dialogs for adding images, articles or categories: based on Vidarsk suggestion modified SQL queries to use CAST(.AS.) instead of CONVERT(.,.) so that same code will work with MySQL and Postgres databases.

- 27.08.14  Merged modifications made by Varlin: Fix resizing problem in Firefox. Fix "size reset" problem: when an image has a defined size, and when you clear the height and width fields, the removal of height and width tags is not remembered (because tags like "_fck_mw_width" are not removed).

- 23.08.14  Merged modifications made by Varlin: integration of texselection plugin which enables to keep selection when switching from WYSIWIG view to source view; changed default table properties; fixed unwanted caps with simple links.

- 18.05.14  Update CKEditor parser to manager calendar tag by oliviermaridat: <calendar></calendar> tags from the Calender extension (http://www.mediawiki.org/wiki/Extension:Calendar_%28Kenyu73%29) are displayed as nowiki balise, and didn't break anymore when the page is saved.

- 11.05.14  Fixed [[Media:<link>|<text>]] internal link to an image or a file of other types. Added checkbox in link.js dialog for 'Media:' -prefix.

- 05.05.14  Fixed focus problem with IE and reference dialog (ref.js) after Ckeditor 4.4.0 upgrade: IE lost focus of ref -element and could not get text of reference into dialog.

- 05.05.14  Updated CKeditor from version 4.3.4 (revision 40ccd20) to version 4.4.0 (revision 98daef5).

- 05.05.14  WYSIWYG extension is compatible with MW 1.22.0 - 1.22.6

- 17.04.14  Add german translation for category and reference buttons/dialogs. Merge pull request #11 from stepping-stone/german-translation-updates.

- 13.04.14  Fixed popup warning of unsaved page (=cancelling edit of page when setting "Warn me when I leave an edit page with unsaved changes" in users preferences has been enabled). Dialog appears only with Cancel- button and is disabled with Save-, Preview- and Diff- buttons. IE required fix also with togglelink.

- 11.04.14  Fixed version info 'Special:Version': WYSIWYG extension 1.5.6_0 [B551], based on this version, modified further, using CKEditor 4.3.4 (revision 40ccd20)

- 10.04.14  Fixed toolbar visibility of WikiEditor in case user preferences defines to start editing with WikiEditor (=wikitext mode).

- 08.04.14  Updated installation of CKeditor 4.3.4 (revision 40ccd20) to include all possible languages provided by CKeditor web site.

- 08.04.14  Merged some of the "French" language modifications made by Varlin.

- 08.04.14  Merged modification "remove <wbr> tags that causes parser to crash" made by Varlin.

- 01.04.14  Updated CKeditor version 4.3.3 to version 4.3.4 (revision 40ccd20).


WYSIWYG/ChangeLog
WYSIWYG extension ChangeLog
===========================

-------------
VERSION 1.5.6_0 [B551+dd.mm.yyyy]
  See README.md of github module "Mediawiki-wysiwyg/WYSIWYG-CKeditor":
     https://github.com/Mediawiki-wysiwyg/WYSIWYG-CKeditor

-------------
VERSION 1.4.2  -> 1.5.5 
  ? ? ?	 
	 
-------------
VERSION 1.4.1


  FEATURES:
    * Any wiki markup that is inserted by the user in the wysiwyg mode of the
      Richtext editor will be escaped when saving the page. Therefore the input
      is not rendered in the wiki page but displayed as it was entered in the
      editor. To use wiki markup and html tags in a wiki page, the user can do
      this in the source mode of the Richtext editor or use the default wiki
      text editor.
    * new button to add the signature into the edited document.
    * Treat the special tags of the Semantic Forms extension not as templates
      but as a "Special Tag" like magic words or variables. The SF special tags
      are 'info', 'for template', 'field', 'end template', 'standard input'. The
      parsing is done only on pages that are in the form namespace and when
      Semantic Forms is installed.
      See also http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13021

  CHANGES:
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13271
      Buttons for Halo features are not hidden when these are not available.
      Instead in the popup dialog an informative text is displayed where the
      user can get the extension to enable the requested feature.
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=10987
      Provide halo autocompletion in the image-picker - if there is no input
      for selecting an image, an image list with the maximum number of results
      as defined in the search limit is returned.
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13792
      External images are handled correcly now. These are rendered in wysiwyg
      mode and the image dialog inserts the correct html element.

  BUGFIXES:
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=14465
      WYSIWYG changes % to ükLR
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=14447
      Wysiwyg editor looses category annotations with empty labels
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13693
      The link attribute in images got lost
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13400
      Links are broken after edition
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13908
      {{{para|}}} is rendered wrong in WYSIWYG
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13894
      IE8: Cannot enter a link, image, table, query, template: popups are not
      displayed (improved bugfix that was done for #13511)
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13795
      Some formating styles like huge fonts are not supported in the wiki
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13852
      Some features of "Numbered lists" in the WYSIWYG editor malfunctioned
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13539
      Language of UI in CKeditor
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13797
      Remove unwanted <style> elements from WYSIWYG text.
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13579
      Edit sections: no wysiwyg-editor appears
      This bug occurred only if $wgCKEditorUrlparamMode was set.
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13775
      Queries are not saved when using WYSIWYG and IE8
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13580
      User-interface of wysiwyg-editor: two languages
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13521
      Creating a template using Special:Createtemplate generates wrong code
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13194
      IE 8 page error (transformation into wikitext didn't work in IE 8)
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13507
      Parse error leads to empty wiki page
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13511
      Query Interface (and other tools) do not open from Wysiwyg in IE8
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13523
      Reopening a query in QI from Wysiwyg shows an empty query
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13540
      Editor changes [[]] into <a href
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13591
      Interwiki links break in WYSIWYG
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=14268
      Open QI from Wysiwyg with query using comparators yields illegal syntax
    * Fix bug http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=14317
      Edit text box disappears when switching between the two text editors


VERSION 1.4.0


  FEATURES:
    * Update from the FCKeditor to the new CKEditor 3.4
    * Applying changes from the official FCKeditor extension (see
      http://www.mediawiki.org/wiki/Extension:FCKeditor_(Official) for details)

  CHANGES:
    * New installation directory is <mw_install_dir>/extensions/WYSIWYG

  BUGFIXES:



VERSION 1.3.2


  FEATURES:

  CHANGES:
    * Query Interface popup has not tabs anymore because in the SMWHalo
      extension 1.5.1 the query interface will support modifying the query
      source directly.
      Use this version only if you have the SMWHalo extension 1.5.1 installed.
    * A define has been added so that other extensions may easily check whether
      the WYSIWYG extension is available or not.

  BUGFIXES:




WYSIWYG/WYSIWYG.php
<?php
/**
 * CKeditor extension - a WYSIWYG editor for MediaWiki
 *
 * @file
 * @ingroup Extensions
 * @version 1.0
 * @author Frederico Caldeira Knabben
 * @author Wiktor Walc <w.walc(at)fckeditor(dot)net>
 * @copyright Copyright © Frederico Caldeira Knabben, Wiktor Walc, other CKeditor team members and other contributors
 * @license GNU Lesser General Public License 2.1 or later
 */
/*
This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Lesser General Public
License as published by the Free Software Foundation; either
version 2.1 of the License, or (at your option) any later version.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public
License along with this library; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
*/

# version of WYSIWYG and CKeditor
define('WYSIWYG_EDITOR_VERSION', '1.5.6_0 [B551+30.10.2015]');
define('CKEDITOR_VERSION',       'CKEditor 4.5.4 (revision d4677a3)');

# Not a valid entry point, skip unless MEDIAWIKI is defined
if( !defined( 'MEDIAWIKI' ) ) {
	echo <<<HEREDOC
To install WYSIWYG extension, put the following line in LocalSettings.php:
require_once( "\$IP/extensions/WYSIWYG/WYSIWYG.php" );
HEREDOC;
	exit( 1 );
}

# quit when comming from the command line,
# special case, to make Halo webtests run (here we don't have a browser)
if ((!array_key_exists('SERVER_NAME', $_SERVER) || $_SERVER['SERVER_NAME'] == NULL) &&
    (strpos($_SERVER['PHP_SELF'], 'run-test.php') === false) )
	return;

// define to check with {{#isExtensionInstalled:wysiwyg}} if extension is installed
// the parser function comes in the SMWHalo extension
define('wysiwyg', 'true}]');

# Configuration variables
// Path to this extension
$wgFCKEditorExtDir = 'extensions/WYSIWYG';

// Path to the actual editor
$wgFCKEditorDir = 'extensions/WYSIWYG/ckeditor/';

$wgFCKEditorToolbarSet = 'Wiki';

// '0' for automatic ('300' minimum).
$wgFCKEditorHeight = '0';

// Array of namespaces that FCKeditor is disabled for. Use constants like NS_MEDIAWIKI here.
$wgFCKEditorExcludedNamespaces = array();

// set this to true if you want the Richeditor show up only when the
// URL param mode=wysiwyg is set
$wgCKEditorUrlparamMode = false;

// hide toolbar buttons when some extensions are not installed (default show buttons)
$wgCKEditorHideDisabledTbutton = false;

/**
 * Enable use of AJAX features.
 */
global $wgAjaxExportList;
require_once('CKeditorSajax.body.php');
$wgAjaxExportList[] = 'wfSajaxSearchImageCKeditor';
$wgAjaxExportList[] = 'wfSajaxSearchArticleCKeditor';
$wgAjaxExportList[] = 'wfSajaxSearchCategoryCKeditor';
$wgAjaxExportList[] = 'wfSajaxWikiToHTML';
$wgAjaxExportList[] = 'wfSajaxGetImageUrl';
$wgAjaxExportList[] = 'wfSajaxGetMathUrl';
$wgAjaxExportList[] = 'wfSajaxSearchTemplateCKeditor';
$wgAjaxExportList[] = 'wfSajaxSearchSpecialTagCKeditor';
$wgAjaxExportList[] = 'wfSajaxToggleCKeditor';

// Extension credits that will show up on Special:Version
$wgExtensionCredits['other'][] = array(
	'path' => __FILE__,
	'name' => 'WYSIWYG editor',
	'author' => array( 'Frederico Caldeira Knabben', 'Wiktor Walc', 'Jack Phoenix', 'ontoprise GmbH', 'MediaWiki community of wysiwyg'),
	'version' => WYSIWYG_EDITOR_VERSION.', '.CKEDITOR_VERSION,
	//'url' => 'http://smwforum.ontoprise.com/smwforum/index.php/Help:WYSIWYG_Extension',
	'url' => 'http://www.mediawiki.org/wiki/Extension:WYSIWYG',
	'descriptionmsg' => 'fckeditor-desc',
);

// Autoloadable classes
global $wgAutoloadClasses;
$dir = dirname( __FILE__ ) . '/';
$wgAutoloadClasses['CKEditor'] = $dir . 'ckeditor/ckeditor_php5.php';
$wgAutoloadClasses['CKeditorParser'] = $dir . 'CKeditorParser.body.php';
$wgAutoloadClasses['CKeditorParserOptions'] = $dir . 'CKeditorParserOptions.body.php';
$wgAutoloadClasses['CKeditorParserWrapper'] = $dir . 'CKeditorParserWrapper.body.php';
$wgAutoloadClasses['CKeditorLinker'] = $dir . 'CKeditorLinker.php';
$wgAutoloadClasses['CKeditorSkin'] = $dir . 'CKeditorSkin.body.php';
$wgAutoloadClasses['CKeditorEditPage'] = $dir . 'CKeditorEditPage.body.php';
$wgAutoloadClasses['CKeditor_MediaWiki'] = $dir . 'CKeditor.body.php';

// Path to internationalization file
$wgExtensionMessagesFiles['CKeditor'] = $dir . 'CKeditor.i18n.php';

// Initialize FCKeditor and the MediaWiki extension
$ckeditor = new CKEditor('fake');
$wgFCKEditorIsCompatible = $ckeditor->IsCompatible();

$oCKeditorExtension = new CKeditor_MediaWiki();

// Hooked functions
global $wgHooks;
$wgHooks['ParserAfterTidy'][]                   = array( $oCKeditorExtension, 'onParserAfterTidy' );
$wgHooks['EditPage::showEditForm:initial'][]    = array( $oCKeditorExtension, 'onEditPageShowEditFormInitial' );
$wgHooks['EditPage::showEditForm:fields'][]		= array( $oCKeditorExtension, 'onEditPageShowEditFormFields' );
$wgHooks['EditPageBeforePreviewText'][]         = array( $oCKeditorExtension, 'onEditPageBeforePreviewText' );
$wgHooks['EditPagePreviewTextEnd'][]            = array( $oCKeditorExtension, 'onEditPagePreviewTextEnd' );
$wgHooks['CustomEditor'][]                      = array( $oCKeditorExtension, 'onCustomEditor' );
$wgHooks['LanguageGetMagic'][]					= 'CKeditor_MediaWiki::onLanguageGetMagic';
$wgHooks['ParserBeforeStrip'][]					= 'CKeditor_MediaWiki::onParserBeforeStrip';
$wgHooks['ParserBeforeInternalParse'][]			= 'CKeditor_MediaWiki::onParserBeforeInternalParse';
$wgHooks['EditPageBeforeConflictDiff'][]		= 'CKeditor_MediaWiki::onEditPageBeforeConflictDiff';
$wgHooks['SanitizerAfterFixTagAttributes'][]	= 'CKeditor_MediaWiki::onSanitizerAfterFixTagAttributes';
$wgHooks['MakeGlobalVariablesScript'][]			= 'CKeditor_MediaWiki::onMakeGlobalVariablesScript';
$wgHooks['GetPreferences'][]					= 'CKeditor_MediaWiki::onGetPreferences';
$wgHooks['DoEditSectionLink'][]					= 'CKeditor_MediaWiki::onDoEditSectionLink';
// bugfix for http://smwforum.ontoprise.com/smwbugs/show_bug.cgi?id=13511
$wgHooks['OutputPageParserOutput'][]            = 'CKeditor_MediaWiki::onOutputPageParserOutput';
$wgHooks['BeforePageDisplay'][]                 = 'CKeditor_MediaWiki::onBeforePageDisplay';
$wgHooks['EditPageBeforeEditButtons'][]         = 'CKeditor_MediaWiki::onEditPageBeforeEditButtons'; // 13.04.14 RL

// Defaults for new preferences options
global $wgDefaultUserOptions;
$wgDefaultUserOptions['riched_use_toggle'] = 1;
$wgDefaultUserOptions['riched_start_disabled'] = 0;
$wgDefaultUserOptions['riched_use_popup'] = 0; // 06.03.15 Varlin 1=>0. Popup is unsupported/untested and will most likely fail with this branch of wysiwyg.
$wgDefaultUserOptions['riched_toggle_remember_state'] = 1;
$wgDefaultUserOptions['riched_link_paste_text'] = 1; // 08.09.14 RL

// when SMWHalo is used then the QueryInterface opens in an Iframe
// also add setting that the Semantic toobar is loaded by default
if (defined('SMW_HALO_VERSION')) {
	global $wgEditPageFrameOptions;
    $wgEditPageFrameOptions = 'SAMEORIGIN';
    $wgDefaultUserOptions['riched_load_semantic_toolbar'] = 1;
}
